public/apps/account/password-reset-panel.tsx

```diff
@@ -0,0 +1,157 @@
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiIcon,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiOverlayMask,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { FormRow } from '../configuration/utils/form-row';
+import { AppDependencies } from '../types';
+import { getUserDetail, updateUser } from '../configuration/utils/internal-user-detail-utils';
+import { InternalUserUpdate } from '../configuration/types';
+import { buildUrl } from '../configuration/utils/url-builder';
+
+interface PasswordResetPanelProps extends AppDependencies {
+  username: string;
+  handleClose: () => void;
+}
+
+export function PasswordResetPanel(props: PasswordResetPanelProps) {
+  const [currentPassword, setCurrentPassword] = useState<string>('');
+  // reply on backend response of login call to verify
+  const [isCurrentPasswordInvalid, setIsCurrentPasswordInvalid] = useState<boolean>(false);
+
+  const [newPassword, setNewPassword] = useState<string>('');
+  // reply on backend response of user update call to verify
+  const [isNewPasswordInvalid, setIsNewPasswordInvalid] = useState<boolean>(false);
+
+  const [repeatNewPassword, setRepeatNewPassword] = useState<string>('');
+  const [isRepeatNewPasswordInvalid, setIsRepeatNewPasswordInvalid] = useState<boolean>(false);
+
+  const handleReset = async () => {
+    // validate the current password
+    try {
+      await props.coreStart.http.post('/auth/login', {
+        body: JSON.stringify({
+          username: props.username,
+          password: currentPassword,
+        }),
+      });
+    } catch (e) {
+      setIsCurrentPasswordInvalid(true);
+      return;
+    }
+
+    // update new password
+    try {
+      const user = await getUserDetail(props.coreStart.http, props.username);
+
+      const updateObject: InternalUserUpdate = {
+        password: newPassword,
+        backend_roles: user.backend_roles,
+        attributes: user.attributes,
+      };
+
+      await updateUser(props.coreStart.http, props.username, updateObject);
+
+      // need to log in with new password
+      window.location.href = buildUrl();
+    } catch (e) {
+      console.error(
+        'Password reset failed, please check console log for more detail. ' + JSON.stringify(e)
+      );
+      setIsNewPasswordInvalid(true);
+    }
+  };
+
+  return (
+    <EuiOverlayMask>
+      <EuiModal onClose={props.handleClose}>
+        <EuiSpacer />
+        <EuiModalBody>
+          <EuiTitle>
+            <h4>Reset password for &quot;{props.username}&quot;</h4>
+          </EuiTitle>
+
+          <EuiSpacer />
+
+          <FormRow
+            headerText="Current password"
+            helpText="Verify your account by entering your current password"
+          >
+            <EuiFieldText
```

Comment:
I saw there is a component named EuiFieldPassword, does that work?

Reply:
Tried EuiFieldPassword. It works. Switch to EuiFieldPassword which is more proper for this context.